### PR TITLE
OSSM-3790 Adding Test cases TestOperator and TestSMMDelete

### PR DESCRIPTION
**Pull request by fjglira**
Jira: https://issues.redhat.com/browse/OSSM-3790

Adding two test cases that was created in 2.3 branch after the creation of the 2.4 branch:

- T39 TestSMMDelete: related to the Jira ticket: https://issues.redhat.com/browse/OSSM-3450
- T40 related to the tickets: https://issues.redhat.com/browse/OSSM-2342 and https://issues.redhat.com/browse/OSSM-3516

Relevant changes besides add two test cases:
UntilSuccessWithOptions: fix order in log message

### CODE REVIEW

**Comment by fjglira** [comment]
Adding label do not merge hold because I need to run subtest Run SMCP Infra Nodes because I was some issues with my cluster

File: pkg/util/oc/pod_commands.go
Comment (luksa) [inline pkg/util/oc/pod_commands.go:131]: What was wrong with the previous message? We use the `<namespace>/<name>` format in several places.

Comment (fjglira) [reply pkg/util/oc/pod_commands.go:131]: Whit the previous message I think it's a bit confusing because we said: `Condition Ready not met smcp istio-system/default` And the command for the wait is `oc wait -n istio-system smcp/basic ...`. If you read the message in this case the one who reads can think that the command is being appliying like `oc wait -n smcp istio-system/default`. What do you think, or maybe we can change the message for something like: `Condition xxxxx not met by kind: xxxx in namespace/name`

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (luksa) [inline pkg/tests/ossm/bug_SMM_delete_test.go:30]: Let's use one of the standard namespaces that the other tests use (foo, bar, legacy, or mesh-external)

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (luksa) [inline pkg/tests/ossm/bug_SMM_delete_test.go:27]: Step too long. Split into multiple steps.

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (luksa) [reply pkg/tests/ossm/bug_SMM_delete_test.go:21]: According to the description, this test should contain two subtests.

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (luksa) [inline pkg/tests/ossm/bug_SMM_delete_test.go:43]:
```suggestion
		t.LogStep("Delete SMM in namespace bookinfo; expect SMMR to be deleted")
```

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (luksa) [inline pkg/tests/ossm/bug_SMM_delete_test.go:51]: Don't use "Success" or "Error" prefix. They are added automatically by the assert function.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:38]: This won't be one test that tests everything about the operator, so you should rename it. This test only tests the deployment of the operator & control plane components on infra nodes, so the test name should reflect this.

Also, remove the comment and replace it with a `t.Log()` which prints the test's purpose/description.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:42]: Move the comment into the Skip string, so that when someone runs this test, it prints why this test was skipped.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:41]: Instead of Smcp.Rosa, you should call env.IsRosa()

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:66]: Why don't you taint in the test setup? That's where this belongs (immediately after you label the node)

File: pkg/util/oc/pod_commands.go
Comment (fjglira) [reply pkg/util/oc/pod_commands.go:131]: Will change this

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (fjglira) [reply pkg/tests/ossm/bug_SMM_delete_test.go:21]: Ok, will make changes here

Comment (fjglira) [reply pkg/tests/ossm/bug_SMM_delete_test.go:21]: I changed the test name to TestSMM and subdivided the test in 3 subtest cases:

- SMM creation creates SMMR
- SMM deletion do not deleted SMMR is the SMMR have more than one namespace
- SMM deletion if the namespace is the only one in the smmr deletes the smmr

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (fjglira) [reply pkg/tests/ossm/bug_SMM_delete_test.go:27]: Done

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (fjglira) [reply pkg/tests/ossm/bug_SMM_delete_test.go:37]: Make some changes here, so I'll not commit this

File: pkg/tests/ossm/bug_SMM_delete_test.go
Comment (fjglira) [reply pkg/tests/ossm/bug_SMM_delete_test.go:51]: Done

File: pkg/tests/ossm/setup.go
Comment (luksa) [inline pkg/tests/ossm/setup.go:103]: Can you move this to its previous position? You effectively just moved this function a few lines down.

File: pkg/util/oc/pod_commands.go
Comment (luksa) [inline pkg/util/oc/pod_commands.go:131]:
```suggestion
				fmt.Sprintf("Condition %s met by %s %s/%s", condition, kind, ns, name),
				fmt.Sprintf("Condition %s not met by %s %s/%s, retrying", condition, kind, ns, name)))
```

File: pkg/tests/ossm/setup.go
Comment (fjglira) [reply pkg/tests/ossm/setup.go:103]: Was a mistake after a merge from upcoming changes. Ok I'll move it

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:40]: This doesn't say anything. Let's explain each test concretely. For example:
```suggestion
		t.Log("This test verifies that the OSSM operator and Istio components can be configured to run on infrastructure nodes")
```

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:42]: This skip message still doesn't explain why the test is being skipped.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:38]:
```suggestion
func TestDeployOnInfraNodes(t *testing.T) {
```

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:43]: Missing skip when cluster only has a single node.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:55]:
```suggestion
		t.NewSubTest("operator").Run(func(t TestHelper) {
```

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:92]:
```suggestion
		t.NewSubTest("control plane").Run(func(t TestHelper) {
```

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [reply pkg/tests/ossm/operator_test.go:55]: This way, the log will show `TestDeployOnInfraNodes/operator` for the operator subtest and `TestDeployOnInfraNodes/control_plane` for the control plane subtest. That's all we need. The t.Log statements then explain each subtest in more detail.

File: pkg/tests/ossm/operator_test.go
Comment (luksa) [inline pkg/tests/ossm/operator_test.go:56]:
```suggestion
			t.Log("Verify OSSM Operator is deployed on infra node when configured")
```

**Review by luksa — CHANGES_REQUESTED** [review]
Did you push all your changes? A lot of my previous comments aren't addressed in the current PR, so I'm just rewriting the same comments I already wrote... that's why I didn't review the entire thing again.

File: pkg/tests/ossm/operator_test.go
Comment (fjglira) [reply pkg/tests/ossm/operator_test.go:42]: Sorry, I was not push all the changes, I was solving some flakiness in the T40

File: pkg/tests/ossm/operator_test.go
Comment (fjglira) [reply pkg/tests/ossm/operator_test.go:38]: I renamed previously to TestRunInInfraNodes, but is better in your way

**Pull request closed without merging**
